### PR TITLE
Added `fastnear` as an alternative provider of NEAR blockchain data

### PR DESCRIPTION
**Pull request by kobayurii**
This PR introduces `fastnear` as an alternative data provider to the `NEAR Lake Framework`. By integrating `fastnear`, we aim to provide a more flexible source for retrieving NEAR blockchain data, enhancing the overall versatility and performance of the framework.

**Changes**

- Extended the existing codebase to support `fastnear` as an additional data source.
- Added necessary configurations and documentation for enabling `fastnear`.
- Implemented mechanisms to switch seamlessly between `NEAR Lake` and `fastnear` providers based on user preference or use case.

**Motivation**

Adding `fastnear` as a source allows users to access NEAR blockchain data from a different provider, potentially benefiting from performance or cost improvements.

### CODE REVIEW

**Comment by kobayurii** [comment]
@race-of-sloths please, include my PR in the Race

**Comment by race-of-sloths** [comment]
@kobayurii Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&pr=near/near-lake-framework-rs/109&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&pr=near/near-lake-framework-rs/109&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/kobayurii)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/kobayurii)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: Cargo.toml
Comment (khorolets) [inline Cargo.toml:39]: Reminder: this is the blocker from merging this change into the 0.7.x branch and releasing a new version

**Review by khorolets — CHANGES_REQUESTED** [review]
Awesome PR that bring a great value to the users and contributes into a decentralization. Great job!

The only thing that is left is around the README, we need to add a section describing that now there is a couple of data providers the framework supports and clearly explain how to use both of them.

Also, we need to add the paragraph about the migration from the previous version even though it's a matter of adding `.into()`.

After that we'll be more ready for merging except for the blocker we have (using the nearcore fork for the near-indexer-primitives crate)

**Review by khorolets — CHANGES_REQUESTED** [review]
Had another closer look on the implementation. Still have some doubts, but don't have any objections. Overall the providers feature is well done. 
Breaking changes are described in the CHANGELOG, and are related to more advanced usage of the framework rather than average. I think we're good to go releasing it. 
We need to check if we're still blocked on the primitives and I requested to provide a link to FastNear's NearData repo in the docs.

Great job!

File: Cargo.toml
Comment (kobayurii) [reply Cargo.toml]: solved

**Comment by kobayurii** [comment]
@race-of-sloths please, include my PR in the Race

**Comment by race-of-sloths** [comment]
### ❓Unknown command 
Please check the command and try again

**Comment by kobayurii** [comment]
@race-of-sloths please, include my PR in the Race

**Comment by race-of-sloths** [comment]
### ❓Unknown command 
Please check the command and try again

**Review by khorolets — APPROVED** [review]
@race-of-sloths score 13

**Pull request merged**
